packages/client/test/cli/cli.spec.ts

```diff
@@ -426,6 +426,30 @@ describe('[CLI]', () => {
     }
     await clientRunHelper(cliArgs, onData)
   }, 30000)
+  //TODO: fix with another test-catching method
+  /*it('should start client with no logger when logLevel is set to off', async () => {
+    const cliArgs = [
+      '--logFile=false',
+      '--logRotate=false',
+      '--logMaxFiles=0',
+      '--logLevelFile="debug"',
+      '--logLevel="off"',
+      '--dev=poa',
+      '--port=39671',
+    ]
+    const onData = async (
+      message: string,
+      child: ChildProcessWithoutNullStreams,
+      resolve: Function,
+    ) => {
+      if (message.includes('logger turned off')) {
+        assert.isTrue(true, 'logLevel option can be used to turn off logger')
+        child.kill()
+        resolve(undefined)
+      }
+    }
+    await clientRunHelper(cliArgs, onData)
+  }, 30000)*/
```

Comment:
Note: this is a newly added test, but this is commented out. The reason is that the tests in this file test stuff by by listening to whatever the output (what is being logged :sweat_smile: ) is. Since this test expects no output, we thus cannot test this, this way.

As a general comment: the way to test these things (like in this file) by listening to comments (and also listening to hardcoded strings, not strings which we read from some file, is not the way to test it. If we change a logging message, we could break a test here because the test expects a certain string.

I think the way to test the CLI like this, and also how the CLI is set up to be tested, is not good. I will open an issue for this.

Reply:
I created this issue: https://github.com/ethereumjs/ethereumjs-monorepo/issues/3983